app/src/components/dialog/Dialog.tsx

```diff
@@ -0,0 +1,107 @@
+import { forwardRef } from "react";
+import {
+  Dialog as AriaDialog,
+  type DialogProps as AriaDialogProps,
+} from "react-aria-components";
+
+import { Button, ButtonProps } from "@phoenix/components/button";
+import { Heading, HeadingProps } from "@phoenix/components/content";
+import { Icon, Icons } from "@phoenix/components/icon";
+import { Flex, FlexProps } from "@phoenix/components/layout";
+import { View, ViewProps } from "@phoenix/components/view";
+
+export type DialogProps = AriaDialogProps;
+
+export const Dialog = forwardRef<HTMLDialogElement, DialogProps>(
+  ({ children, ...props }, ref) => {
+    return (
+      <AriaDialog {...props} ref={ref}>
+        {children}
+      </AriaDialog>
+    );
+  }
+);
+
+Dialog.displayName = "Dialog";
+
+export type DialogContentProps = FlexProps;
+
+export const DialogContent = ({ children, ...props }: DialogContentProps) => {
+  return (
+    <Flex direction="column" height="100%" {...props}>
+      {children}
+    </Flex>
+  );
+};
+
+export type DialogHeaderProps = ViewProps;
```

Comment:
minor nit: it will be easier to find these components if in a separate file long term